### ISSUE TITLE
Resolvers needs to be assigned to rather than appended to

### ISSUE THREAD

**Issue by nugend** [sbt/sbt#1138]
I encountered an issue where SBT could not resolve a dependency from a custom resolver setting unless I overrode the resolvers setting key rather than appending, that is:

``` sbt
resolvers += "JBoss" at "https://repository.jboss.org/nexus/content/groups/public"
```

would not work, but

``` sbt
resolvers := Seq("JBoss" at "https://repository.jboss.org/nexus/content/groups/public")
```

did work.

Since resolvers is empty in a fresh project, this seems not to make any sense.

I am using SBT 0.13.1 and Scala 2.10.3

Please see this SO question I wrote for additional details regarding the errors I encountered while running SBT: http://stackoverflow.com/questions/22029393/why-is-sbt-unable-to-resolve-my-dependency/22049771#22049771
As well as this SO question which is how I figured out to use assignment rather than append:http://stackoverflow.com/questions/15593715/sbt-wont-resolve-transitive-dependency-against-any-repository-except-typesafes?rq=1

Also, if it helps, here is a link to the non-working state of the project: nugend/jmsScala@c3860e8
And the working state: nugend/jmsScala@7beedd2

**Comment by jsuereth** [comment]
What is the artifact?  There's an issue with Ivy where, if it find a pom.xml or ivy.xml file in a resolver _AND_ that resolver does not contain the jar, then it will fail to resolve the artifact _even if_ another resolver has it.  It's a known ivy limitation that we'd love to workaround. 

From your output, it looks like somehow you're successfully downloading the POM from maven central and then failing to find the JAR, so this looks related.  Another workaround is to append your jboss resolver so it's FIRST in the list (dangerous as you'll hit this before any local resolvers):

``` scala
fullResolvers := {
  <my jboss resolver>  +: fullResolvers.value
}
```

It's also possible that your ivy cache itself is corrupted.  There was an issue in older sbt versions where resolver names were stored for artifacts.  Unfortunately, this is hard to detect and can hurt new sbt versions.  If this is the case, you can "fix" the issue by removing the area of your `~/.ivy2/cache` directory related to the troublesome artifacts.

Let me know if you still have issues.

**Comment by nugend** [comment]
The artifact is the JMS jar. Maven central doesn't have it (but it does have the POM), JBoss does.

Hmm, I tried retesting this with a completely clean ivy cache and hit the error. I have no clue why changing to assignment fixed the issue. I'm seeing some inconsistent behavior in trying to clearly reproduce why the JMS jar gets downloaded or not sometimes (I'm deleting the javax.jms ivy2 cache and retrying and sometimes it hits JBoss and sometimes it doesn't). I guess I'll try setting fullResolvers  since I need this to work consistently. Unfortunately, I don't have a whole lot of time to mess around with this.

Would filing a bug upstream with ivy help at all? Or is that expected behavior?

edit: Actually, this is a silly idea, but maybe a workaround could be automated inside sbt? If the resolution through the normal list of resolvers doesn't work after a first pass, maybe a second pass through ivy could be attempted with the default resolvers removed. That's a real nasty hack though.

**Comment by jsuereth** [comment]
ivy sees very little movement, so opening an issue is probably the right thing to do, but I wouldn't expect movement. I think that's how Ivy is intended to behave.

For sbt, you want to do the fullowing:

1).  Run `show fullResolvers` and make sure the jboss repository shows up BEFORE any nusance repositories (like, apparently, maven central).

2) If the resolver list is not showing up correctly, then modify it via the:

``` scala
fullResolvers := {
  <the one I need first> +: resolvers.value
}
```

You can also just transform the list of the resolvers in place.

Basically, what you want is the ORDER of the resolvers to be such that the "good" resolver is seen before any "bad' resolver, and that should fix your issue.  It's a known issue with using Ivy, we just haven't figured out an appropriate workaround on our end, nor do I think Ivy would "fix" their side.

**Comment by eed3si9n** [comment]
I'm going to classify this as a bug for now since it's sbt that chose Ivy. We are probably not going to fix this any time soon, but it's a bug.

If the ordering of the resolver matters, could the following work as a workaround?

``` scala
resolvers := ("JBoss" at "https://repository.jboss.org/nexus/content/groups/public") +: resolvers.value
```

**Comment by jsuereth** [comment]
@eed3si9n That workaround almost works.  Maven-central may be contributed from the "boot" repositories (those used by the sbt launcher), which is why you need to attack `fullResolvers` not `resolvers`.

**Comment by eed3si9n** [comment]
Since there's a workaround for this, I'm going to close this as won't fix.

**Comment by SteveSmithTech** [comment]
Hi @jsuereth and @eed3si9n, just to say thanks for the above workaround. We hit this problem with Ivy this week and your workaround has resolved it
